code/__HELPERS/areas.dm

```diff
@@ -0,0 +1,86 @@
+// Gets an atmos isolated contained space
+// Returns an associative list of turf|dirs pairs
+// The dirs are connected turfs in the same space
+// break_if_found is a typecache of turf types to return false if found
+/proc/detect_room(turf/origin, list/break_if_found)
+	if(origin.blocks_air)
+		return list(origin)
+
+	. = list()
+	var/list/checked_turfs = list()
+	var/list/found_turfs = list(origin)
+	while(found_turfs.len)
+		var/turf/sourceT = found_turfs[1]
+		if(break_if_found[sourceT.type])
+			return FALSE
+		found_turfs.Cut(1, 2)
+		var/dir_flags = checked_turfs[sourceT]
+		for(var/dir in GLOB.alldirs)
+			if(dir_flags & dir) // This means we've checked this dir before, probably from the other turf
+				continue
+			var/turf/checkT = get_step(sourceT, dir)
+			if(!checkT)
+				continue
+			checked_turfs[sourceT] |= dir
+			checked_turfs[checkT] |= turn(dir, 180)
+			.[sourceT] |= dir
+			.[checkT] |= turn(dir, 180)
+			var/static/list/diagonal_cache = GLOB.diagonals
+			if((dir in diagonal_cache) || checkT.blocks_air || !CANATMOSPASS(sourceT, checkT))
+				continue
+			found_turfs += checkT // Since checkT is connected, add it to the list to be processed
+
+/proc/create_area(mob/creator)
+	var/static/blacklisted_turfs = typecacheof(/turf/open/space)
+	var/list/turfs = detect_room(get_turf(creator), blacklisted_turfs)
+	if(!turfs)
+		to_chat(creator, "<span class='warning'>The new area must be completely airtight.</span>")
+		return
+
+	var/list/areas = list()
+	var/list/nice_areas = list("New area")
+	for(var/i in 1 to turfs.len)
+		var/area/place = get_area(turfs[i])
+		nice_areas |= place.name
```

Comment:
our survey says: no

Reply:
I hadn't actually used input yet and was going to test if it accepted assoc lists but forgot. I've got some other minor stuff I should probably cleanup as well while I'm looking at this bit of code that I'll handle tomorrow.

Reply:
It does, it just uses the keys of the assoc list as the options' display text